cmd/restic/cmd_ls.go

```diff
@@ -61,8 +65,81 @@ func runLs(opts LsOptions, gopts GlobalOptions, args []string) error {
 
 	ctx, cancel := context.WithCancel(gopts.ctx)
 	defer cancel()
+
+	return printLS(ctx, repo, gopts, opts, args)
+}
+
+type lsSnapshot struct {
+	*restic.Snapshot
+
+	ID      *restic.ID `json:"id"`
+	ShortID string     `json:"short_id"`
+	Nodes   []lsNode   `json:"nodes"`
+}
+
+type lsNode struct {
+	Name       string      `json:"name"`
+	Type       string      `json:"type"`
+	Path       string      `json:"path"`
+	Size       uint64      `json:"size,omitempty"`
+	Mode       os.FileMode `json:"mode,omitempty"`
+	ModTime    time.Time   `json:"mtime,omitempty"`
+	AccessTime time.Time   `json:"atime,omitempty"`
+	ChangeTime time.Time   `json:"ctime,omitempty"`
```

Comment:
I seem to recall that `omitempty` doesn't work with the zero `time.Time` values -- has that changed?

Reply:
Don't know, copied from restic.Node, I'm just beginning with go